src/Native/Unix/System.Native/pal_io.c

```diff
@@ -33,7 +33,10 @@
 #elif HAVE_SENDFILE_4
 #include <sys/sendfile.h>
 #endif
-#if HAVE_INOTIFY
+#if defined(__FreeBSD__) & defined(HAVE_INOTIFY)
```

Comment:
This should not be defined(HAVE_NOTIFY), but rather just HAVE_INOTIFY. The cmake configuration always define that symbol, only its value indicates whether the config time check succeeded or not.